libnetwork/portmapper/proxy_linux.go

```diff
@@ -61,6 +68,22 @@ func (p *proxyCommand) Start() error {
 	}
 	w.Close()
 
+	go func() {
+		p.waitMu.Lock()
+		if p.waitCond.L == nil {
+			p.waitCond.L = &p.waitMu
+		}
+		p.waitError = p.cmd.Wait()
+		p.waitCond.Broadcast()
+		p.exitedWait = true
+		p.waitMu.Unlock()
+		if p.waitError != nil {
+			logrus.Warnf("docker-proxy pid:%d run error: %v", p.cmd.Process.Pid, p.waitError)
+		} else {
+			logrus.Infof("docker-proxy pid:%d exited successfully", p.cmd.Process.Pid)
+		}
```

Comment:
Slightly on the fence if we need the `Info` logs for the "success" case (daemon logs already can be quite noisy); I personally think we could remove it (or if we really think it's useful _at most_ should be a `logrus.Debug()`